lib/smart_answer_flows/marriage-abroad.rb

```diff
@@ -126,6 +126,13 @@ def define
           elsif calculator.partner_is_opposite_sex?
             if calculator.ceremony_country == 'hong-kong'
               outcome :outcome_opposite_sex_marriage_in_hong_kong
+            elsif calculator.ceremony_country == 'south-africa' &&
+                (
+                  calculator.resident_of_uk? ||
+                  calculator.resident_of_ceremony_country? ||
+                  calculator.resident_of_third_country?
+                )
```

Comment:
@leenagupte this check initially started with like with the require for user either a uk resident or resident of ceremony_country

```
elsif calculator.ceremony_country == 'south-africa' &&
      (
         calculator.resident_of_uk? ||
         calculator.resident_of_ceremony_country? 
      )
```

After factcheck FCO made another requirement for third_country

I have decided to keep it this way for explicit reasons, make it a bit more obvious the 3 potential areas affected by this node.